### PR TITLE
adapt to mirah 0.1.3 changes using Mirac.compile. Fix test mojo phase an...

### PR DESCRIPTION
**Pull request by uujava**
Modified mirah maven plugin to:
- use latest 0.1.3 mirah
- use Mirah.new.compile instead of MirahCommand
- bumped version to 1.2
- source generation mojo is still present, but is not working as --java switch not supported

### CODE REVIEW

**Comment by uujava** [comment]
Is it also possible to publish new maven-mirah-plugin version to maven to public repository?

**Comment by baroquebobcat** [comment]
I think @calavera has rights to push new artifacts. We should widen that a
bit though.

On Wed, Mar 18, 2015 at 5:01 AM, uujava <notifications@github.com> wrote:

> Is it also possible to publish new maven-mirah-plugin version to maven to
> public repository?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mirah/maven-mirah-plugin/pull/8#issuecomment-82898924>
> .
>



-- 
-Nick Howard
http://blog.baroquebobcat.com/

**Comment by calavera** [comment]
I'm going to be really honest, I don't know how or where to do that anymore :smile: 

If I remember right, I pushed the artifact to oss.sonatype.org, but I just logged in there and I don't see anywhere how to manage repositories.

I think it would be best if you contact the administers there and ask them to replace my credentials with yours. Feel free to tell them to contact me at david.calavera at gmail. That's the email address I used to sign up.

File: pom.xml [hunk 1]
```diff
@@ -65,7 +65,7 @@
     <dependency>
       <groupId>org.mirah</groupId>
       <artifactId>mirah-complete</artifactId>
-      <version>0.1.1</version>
+      <version>0.1.3</version>
```
Comment (baroquebobcat) [inline pom.xml]: you might as well bump up to 0.1.4

**Comment by baroquebobcat** [comment]
Could you update the formatting to match the existing? ie spaces not tabs, 4 spaces per indent, space after parens before curly braces in if statements / method definitions.

File: pom.xml (hunk 1 above)
Comment (uujava) [inline pom.xml]: Do we have mirah 0.1.4 released and published?
I have not found it in public maven repo. There is 0.1.3 still latest the version: http://mvnrepository.com/artifact/org.mirah/

Comment (uujava) [inline pom.xml]: Mirah 0.1.4  version could be set in plugin configuration. I'm using following plugin config with 0.1.4 published in my local artifactory:
```xml
<plugin>
                    <groupId>org.mirah.maven</groupId>
                    <artifactId>maven-mirah-plugin</artifactId>        
                    <version>1.2-SNAPSHOT</version>				  
                    <executions>
                        <execution>
                            <id>mirah-compile</id>
                            <phase>compile</phase>
                            <goals>
                                <goal>compile</goal>
                            </goals>
                        </execution>
                        <execution>
                            <id>mirah-test-compile</id>
                            <phase>test-compile</phase>
                            <goals>
                                <goal>testCompile</goal>
                            </goals>
                        </execution>
                    </executions>
                    <configuration>
                        <verbose>false</verbose>
                        <newClosures>false</newClosures>
                    </configuration>
					<dependencies>
					  <dependency>
						<groupId>org.mirah</groupId>
						<artifactId>mirah-complete</artifactId>
						<version>0.1.4</version>
					  </dependency>
					</dependencies>
                </plugin>
```

Comment (baroquebobcat) [inline pom.xml]: Hmm. I thought I'd published it. I must have missed a step.

On Mon, Mar 23, 2015 at 3:02 AM, uujava <notifications@github.com> wrote:

> In pom.xml
> <https://github.com/mirah/maven-mirah-plugin/pull/8#discussion_r26920938>:
>
> > @@ -65,7 +65,7 @@
> >      <dependency>
> >        <groupId>org.mirah</groupId>
> >        <artifactId>mirah-complete</artifactId>
> > -      <version>0.1.1</version>
> > +      <version>0.1.3</version>
>
> Mirah 0.1.4 version could be set in plugin configuration. I'm using
> following plugin config with 0.1.4 published in my local artifactory:
>
> <plugin>
>                     <groupId>org.mirah.maven</groupId>
>                     <artifactId>maven-mirah-plugin</artifactId>
>                     <version>1.2-SNAPSHOT</version>
>                     <executions>
>                         <execution>
>                             <id>mirah-compile</id>
>                             <phase>compile</phase>
>                             <goals>
>                                 <goal>compile</goal>
>                             </goals>
>                         </execution>
>                         <execution>
>                             <id>mirah-test-compile</id>
>                             <phase>test-compile</phase>
>                             <goals>
>                                 <goal>testCompile</goal>
>                             </goals>
>                         </execution>
>                     </executions>
>                     <configuration>
>                         <verbose>false</verbose>
>                         <newClosures>false</newClosures>
>                     </configuration>
>                     <dependencies>
>                       <dependency>
>                         <groupId>org.mirah</groupId>
>                         <artifactId>mirah-complete</artifactId>
>                         <version>0.1.4</version>
>                       </dependency>
>                     </dependencies>
>                 </plugin>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mirah/maven-mirah-plugin/pull/8/files#r26920938>.
>



-- 
-Nick Howard
http://blog.baroquebobcat.com/

**Comment by baroquebobcat** [comment]
I pushed 0.1.4 to maven again, correctly I think.

File: pom.xml (hunk 1 above)
Comment (baroquebobcat) [inline pom.xml]: It's now up!

On Mon, Mar 23, 2015 at 7:42 AM, Nick Howard <ndh@baroquebobcat.com> wrote:

> Hmm. I thought I'd published it. I must have missed a step.
>
> On Mon, Mar 23, 2015 at 3:02 AM, uujava <notifications@github.com> wrote:
>
>> In pom.xml
>> <https://github.com/mirah/maven-mirah-plugin/pull/8#discussion_r26920938>
>> :
>>
>> > @@ -65,7 +65,7 @@
>> >      <dependency>
>> >        <groupId>org.mirah</groupId>
>> >        <artifactId>mirah-complete</artifactId>
>> > -      <version>0.1.1</version>
>> > +      <version>0.1.3</version>
>>
>> Mirah 0.1.4 version could be set in plugin configuration. I'm using
>> following plugin config with 0.1.4 published in my local artifactory:
>>
>> <plugin>
>>                     <groupId>org.mirah.maven</groupId>
>>                     <artifactId>maven-mirah-plugin</artifactId>
>>                     <version>1.2-SNAPSHOT</version>
>>                     <executions>
>>                         <execution>
>>                             <id>mirah-compile</id>
>>                             <phase>compile</phase>
>>                             <goals>
>>                                 <goal>compile</goal>
>>                             </goals>
>>                         </execution>
>>                         <execution>
>>                             <id>mirah-test-compile</id>
>>                             <phase>test-compile</phase>
>>                             <goals>
>>                                 <goal>testCompile</goal>
>>                             </goals>
>>                         </execution>
>>                     </executions>
>>                     <configuration>
>>                         <verbose>false</verbose>
>>                         <newClosures>false</newClosures>
>>                     </configuration>
>>                     <dependencies>
>>                       <dependency>
>>                         <groupId>org.mirah</groupId>
>>                         <artifactId>mirah-complete</artifactId>
>>                         <version>0.1.4</version>
>>                       </dependency>
>>                     </dependencies>
>>                 </plugin>
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/mirah/maven-mirah-plugin/pull/8/files#r26920938>.
>>
>
>
>
> --
> -Nick Howard
> http://blog.baroquebobcat.com/
>
>


-- 
-Nick Howard
http://blog.baroquebobcat.com/

**Comment by uujava** [comment]
updated plugin dependency to mirah-complete 0.1.4. tested.

**Pull request merged**